Kentico.Kontent.Delivery/ContentItems/ModelProvider.cs

```diff
@@ -220,7 +225,7 @@ private async Task<object> GetPropertyValueAsync(JObject elementsData, PropertyI
                     return (elementValue["type"].ToString()) switch
                     {
                         "rich_text" => await GetElementModelAsync<RichTextElementValue, string>(property, context, elementValue, valueConverter),
-                        "asset" => await GetElementModelAsync<ContentElementValue<Asset>, Asset>(property, context, elementValue, valueConverter),
```

Comment:
I do not think this could have ever worked. Asset element always contains list of assets.